### ISSUE TITLE
creating new ffmpeginteropConfig instance throws exception

### ISSUE THREAD

**Issue by touseefbsb** [lukasf/FFmpegInterop#67]
**BadImageFormatException**:  is not a valid Win32 application. (Exception from HRESULT: 0x800700C1)

**StackTrace** : "   at System.StubHelpers.StubHelpers.GetWinRTFactoryObject(IntPtr pCPCMD)\r\n   at FFmpegInterop.FFmpegInteropConfig..ctor()\r\n   at Fluent_Video_Player.Helpers.FluentPlaybackList..ctor()\r\n   at Fluent_Video_Player.Helpers.MainMediaPlayer..ctor()\r\n   at Fluent_Video_Player.Helpers.MediaHelper..cctor()"

the same code was working fine before but I am rebuilding my app from scratch in a new project with min target as **1809** and I am using same code for media services, and now when I create a new instance of config class it throws above described exception. ( **Config = new FFmpegInteropConfig();** )

**Comment by lukasf** [comment]
I had that weird exception myself, but unfortunately I don't remember why or how it went away. You might have to update that NETCore uwp nuget packet to latest version. Try Clean+Rebuild. Also make sure build platforms are same for all modules. Maybe update build platform of FFmpegInterop to 1809 as well. Good luck. I know it took me quite some time to get this fixed.

**Comment by touseefbsb** [comment]
tried all this with zero success, I also tried the simple constructor, i.e  : without creating a config object at all, but still get the same exception.

**Comment by khouzam** [comment]
Can you set the logging settings to detailed and share your log?

**Comment by touseefbsb** [comment]
are you talking about MSBuild logging? where would I get the output file for that after build? @khouzam

**Comment by touseefbsb** [comment]
btw for some unknown reason ffmpeginterop project now only build if it is the only loaded project in the solution and client app is not even trying to build even when being the only project being loaded in the solution, this has happened before as well but I guess it is unrelated to this.

**Comment by touseefbsb** [comment]
@khouzam  build log is very large, so I pasted it in a textfile and attaching it here. it contains build of ffmpeginterop project and then build of my client app project as well. 
[BuildLogs.txt](https://github.com/lukasf/FFmpegInterop/files/2499269/BuildLogs.txt)

**Comment by khouzam** [comment]
Hi @touseefbsb,

It looks like for some reason, the `ARM` version of FFmpeg libraries are being included in your project which explains the `not a valid win32 app` error message.

You should double check both your environment and the path to the libraries that you've set in your project.

```
2>  Fluent_Video_Player : Fluent_Video_Player.ContentFilesProjectOutputGroup : C:\Users\fcuv3\Source\Repos\TouseefToolkit\TouseefToolkitSolution\ffmpeg\Build\Windows10\ARM\bin\avcodec-57.dll -> avcodec-57.dll
2>Done executing task "Message".
2>Task "Message"
2>  Fluent_Video_Player : Fluent_Video_Player.ContentFilesProjectOutputGroup : C:\Users\fcuv3\Source\Repos\TouseefToolkit\TouseefToolkitSolution\ffmpeg\Build\Windows10\ARM\bin\avdevice-57.dll -> avdevice-57.dll
2>Done executing task "Message".
2>Task "Message"
2>  Fluent_Video_Player : Fluent_Video_Player.ContentFilesProjectOutputGroup : C:\Users\fcuv3\Source\Repos\TouseefToolkit\TouseefToolkitSolution\ffmpeg\Build\Windows10\ARM\bin\avfilter-6.dll -> avfilter-6.dll
2>Done executing task "Message".
2>Task "Message"
2>  Fluent_Video_Player : Fluent_Video_Player.ContentFilesProjectOutputGroup : C:\Users\fcuv3\Source\Repos\TouseefToolkit\TouseefToolkitSolution\ffmpeg\Build\Windows10\ARM\bin\avformat-57.dll -> avformat-57.dll
2>Done executing task "Message".
2>Task "Message"
2>  Fluent_Video_Player : Fluent_Video_Player.ContentFilesProjectOutputGroup : C:\Users\fcuv3\Source\Repos\TouseefToolkit\TouseefToolkitSolution\ffmpeg\Build\Windows10\ARM\bin\avutil-55.dll -> avutil-55.dll
2>Done executing task "Message".
2>Task "Message"
2>  Fluent_Video_Player : Fluent_Video_Player.ContentFilesProjectOutputGroup : C:\Users\fcuv3\Source\Repos\TouseefToolkit\TouseefToolkitSolution\ffmpeg\Build\Windows10\ARM\bin\swresample-2.dll -> swresample-2.dll
2>Done executing task "Message".
2>Task "Message"
2>  Fluent_Video_Player : Fluent_Video_Player.ContentFilesProjectOutputGroup : C:\Users\fcuv3\Source\Repos\TouseefToolkit\TouseefToolkitSolution\ffmpeg\Build\Windows10\ARM\bin\swscale-4.dll -> swscale-4.dll
2>Done executing task "Message".
```

**Comment by touseefbsb** [comment]
@khouzam 
actually I have total 3 projects in my solution. 
1. ffmpeginterop project
2. old client app
3. new client app.

this build log was from the new client app I am working on, and the path to the included dll files are exactly the same as old project, because I literally copied and pasted these references from old client project to the new one, that is why both are able to find the path correctly and build. But wht confuses me is tht the old client and also the sample project ( in differently repo and different folder altogether ) are giving the class not registered exception on 1809 but this project gives a different one.
But anyways how can I assure that the referenced paths in my new client project are correct?

**Comment by khouzam** [comment]
I don't see the class not registered on the project, at least not from the samples. I'm on a build past 1809 but I can try this on 1809 tonight.

For the wrong architecture, you should be using a similar model that is in the samples. For example looking at `FFmpegInterop\Samples\SamplesWin10\MediaPlayerCS\MediaPlayerCS.csproj`

```XML
    <Content Include="$(SolutionDir)ffmpeg\Build\Windows10\$(PlatformTarget)\bin\avcodec-57.dll" />
    <Content Include="$(SolutionDir)ffmpeg\Build\Windows10\$(PlatformTarget)\bin\avdevice-57.dll" />
    <Content Include="$(SolutionDir)ffmpeg\Build\Windows10\$(PlatformTarget)\bin\avfilter-6.dll" />
    <Content Include="$(SolutionDir)ffmpeg\Build\Windows10\$(PlatformTarget)\bin\avformat-57.dll" />
    <Content Include="$(SolutionDir)ffmpeg\Build\Windows10\$(PlatformTarget)\bin\avutil-55.dll" />
    <Content Include="$(SolutionDir)ffmpeg\Build\Windows10\$(PlatformTarget)\bin\swresample-2.dll" />
    <Content Include="$(SolutionDir)ffmpeg\Build\Windows10\$(PlatformTarget)\bin\swscale-4.dll" />
```

**Comment by touseefbsb** [comment]
@khouzam  thanks, you were correct, the dll references were not correct, I was able to fix the win32 exception with correct reference, but now I am getting the same exception in my new client app as the old one and as the sampleApp one **Class is not registered**

**Comment by touseefbsb** [comment]
Thanks a lot for all the help guys, I was finally able to fix both exceptions discussed here.

1. **not a valid win32 app exception** : as explained by @khouzam  above, project should have correct path to the dll files to fix this exception.
2. **Class Not Registered** : Basically my main testing device was an Azure VM which had Windows 10 version N, means it has no Media capabilities pre-installed, and since 1809 was pulled back and not officially re-released yet, I couldnt install a MediaPack for this version yet, and most likely that was causing this exception on FFmpegInterop related initialization of objects. So I tested on my own laptop again, after cleaning and rebuilding, and this time I got no exception because my own laptop does have 1809 build ( from insider slow ring ) and it is Windows 10 pro version which does have media features pre-installed.
